pyface/ui/qt4/data_view/data_view_widget.py

```diff
@@ -21,29 +25,162 @@
+}
+pyface_selection_types = {
+    value: key for key, value in qt_selection_types.items()
+}
+qt_selection_modes = {
+    "none": QAbstractItemView.NoSelection,
+    "single": QAbstractItemView.SingleSelection,
+    "extended": QAbstractItemView.ExtendedSelection,
+}
+pyface_selection_modes = {
+    value: key for key, value in qt_selection_modes.items()
+}
+
 
 @provides(IDataViewWidget)
 class DataViewWidget(MDataViewWidget, Widget):
     """ The Qt implementation of the DataViewWidget. """
 
+    # IDataViewWidget Interface traits --------------------------------------
+
+    #: What can be selected.  Qt supports additional selection types.
+    selection_type = Enum("row", "column", "item")
+
+    #: How selections are modified.  Qt supports turning off selections.
+    selection_mode = Enum("extended", "none", "single")
+
+    # IWidget Interface traits ----------------------------------------------
+
+    control = Instance(QAbstractItemView)
+
+    # Private traits --------------------------------------------------------
+
     #: The QAbstractItemModel instance used by the view.  This will
     #: usually be a DataViewItemModel subclass.
     _item_model = Instance(QAbstractItemModel)
 
+    # ------------------------------------------------------------------------
+    # IDataViewWidget Interface
+    # ------------------------------------------------------------------------
+
+    def _create_item_model(self):
+        """ Create the DataViewItemModel which wraps the data model. """
+        self._item_model = DataViewItemModel(self.data_model)
+
+    def _get_control_header_visible(self):
+        """ Method to get the control's header visibility. """
+        return not self.control.isHeaderHidden()
+
+    def _set_control_header_visible(self, header_visible):
+        """ Method to set the control's header visibility. """
+        self.control.setHeaderHidden(not header_visible)
+
+    def _get_control_selection_type(self):
+        """ Toolkit specific method to get the selection type. """
+        qt_selection_type = self.control.selectionBehavior()
+        return pyface_selection_types[qt_selection_type]
+
+    def _set_control_selection_type(self, selection_type):
+        """ Toolkit specific method to change the selection type. """
+        qt_selection_type = qt_selection_types[selection_type]
+        self.control.setSelectionBehavior(qt_selection_type)
+        self.selection = []
+
+    def _get_control_selection_mode(self):
+        """ Toolkit specific method to get the selection mode. """
+        qt_selection_mode = self.control.selectionMode()
+        return pyface_selection_modes[qt_selection_mode]
+
+    def _set_control_selection_mode(self, selection_mode):
+        """ Toolkit specific method to change the selection mode. """
+        qt_selection_mode = qt_selection_modes[selection_mode]
+        self.control.setSelectionMode(qt_selection_mode)
+        self.selection = []
```

Comment:
I agree that shifting the selection setting up to the mixin is a good idea.

I don't see how it will allow avoiding the branch in the test, since `self.assertEqual(self.widget._get_control_selection_mode(), "none")` will fail if the toolkit is wx.

Reply:
Shifting to the mixin breaks Wx, so I've reverted this.